crypto.go

```diff
@@ -637,12 +696,12 @@ func (s chachaPolyScheme) New(key []byte) (cipher.AEAD, error) {
 	return chacha20poly1305.New(key)
 }
 
-func (s chachaPolyScheme) KeySize() int {
-	return chacha20poly1305.KeySize
+func (s chachaPolyScheme) KeySize() uint16 {
+	return uint16(chacha20poly1305.KeySize)
 }
 
-func (s chachaPolyScheme) NonceSize() int {
-	return chacha20poly1305.NonceSize
+func (s chachaPolyScheme) NonceSize() uint16 {
```

Comment:
It seems like these `XSize()` methods should either all be `uint16` or all be `int`.  My preference would be to stay with the latter, since their primary usage is with things like `make()`, where Go uses `int` for sizes.

Reply:
That works for me! I was trying to remove casts and checks, but we only really check in one place anyway (when encoding the length in LabeledExtract).